test/src/test/java/hudson/model/ComputerTest.java

```diff
@@ -52,4 +66,27 @@ public void discardLogsAfterDeletion() throws Exception {
 
         assertTrue("Slave log should be kept", keep.toComputer().getLogFile().exists());
     }
+
+    /**
+     * Verify we can't rename a node over an existing node.
+     */
+    @Issue("JENKINS-31321")
+    @Test
+    public void testProhibitRenameOverExistingNode() throws Exception {
+        final String NOTE = "Rename node to name of another node should fail.";
+
+        Node nodeA = j.createSlave("nodeA", null, null);
+        Node nodeB = j.createSlave("nodeB", null, null);
+
+        WebClient wc = j.createWebClient();
+        HtmlForm form = wc.getPage(nodeB, "configure").getFormByName("config");
+        form.getInputByName("_.name").setValueAttribute("nodeA");
+
+        try {
+            j.submit(form);
+            fail(NOTE);
+        } catch (FailingHttpStatusCodeException e) {
+            assertThat(NOTE, e.getStatusCode(), equalTo(400));
```

Comment:
Makes sense to also check the error message. Just to be sure the operation fails due to this issue